flake.nix

```diff
@@ -1,206 +1,106 @@
-    rv-utils.url = "github:runtimeverification/rv-nix-tools";
+    # pin rv-utils to a fixed version
+    rv-utils.url = "github:runtimeverification/rv-nix-tools/854d4f0";
     nixpkgs.follows = "rv-utils/nixpkgs";
-    stacklock2nix.url = "github:cdepillabout/stacklock2nix";
     z3 = {
       url = "github:Z3Prover/z3/z3-4.13.4";
       flake = false;
     };
+    flake-utils.url = "github:numtide/flake-utils";
   };
-  outputs = { self, nixpkgs, stacklock2nix, z3, rv-utils }:
-    let
-      perSystem = nixpkgs.lib.genAttrs nixpkgs.lib.systems.flakeExposed;
-      nixpkgsCleanFor = system: import nixpkgs { inherit system; };
-      nixpkgsFor = system:
-        import nixpkgs {
-          inherit system;
-          overlays = [ stacklock2nix.overlay self.overlay self.overlays.z3 ];
-        };
-      withZ3 = pkgs: pkg: exe:
-        pkgs.stdenv.mkDerivation {
-          name = exe;
-          phases = [ "installPhase" ];
-          buildInputs = with pkgs; [ makeWrapper ];
-          installPhase = ''
-            mkdir -p $out/bin
-            makeWrapper ${pkg}/bin/${exe} $out/bin/${exe} --prefix PATH : ${pkgs.z3}/bin
-          '';
-        };
-      # This should based on the compiler version from the resolver in stack.yaml.
-      ghcVersion = pkgs: pkgs.haskell.packages.ghc965;
-    in {
-      overlay = final: prev: {
-        haskell-backend = final.stacklock2nix {
-          stackYaml = ./stack.yaml;
-          # This should based on the compiler version from the resolver in stack.yaml.
-          baseHaskellPkgSet = ghcVersion final;
-          cabal2nixArgsOverrides = args:
-            args // {
-              # The Haskell package `"graphviz"` depends on the _system_
-              # package `graphviz`, and takes the system package `graphviz` as one of its build
-              # inputs, but it is actually getting passed _itself_ (not the system package
-              # `graphviz`), which causes the infinite recursion.
-              "graphviz" = _: { graphviz = final.graphviz; };
-            };
-          additionalHaskellPkgSetOverrides = hfinal: hprev:
-            with final.haskell.lib; {
-              crypton-x509 = dontCheck hprev.crypton-x509;
-              data-fix = doJailbreak hprev.data-fix;
-              decision-diagrams = dontCheck hprev.decision-diagrams;
-              fgl = dontCheck hprev.fgl;
-              fgl-arbitrary = dontCheck hprev.fgl-arbitrary;
-              graphviz = dontCheck hprev.graphviz;
-              json-rpc = dontCheck hprev.json-rpc;
-              lifted-base = dontCheck hprev.lifted-base;
-              prettyprinter = dontCheck hprev.prettyprinter;
-              semialign = doJailbreak hprev.semialign;
-              smtlib-backends-process = dontCheck hprev.smtlib-backends-process;
-              tar = dontCheck hprev.tar;
-              text-short = doJailbreak hprev.text-short;
-              these = doJailbreak hprev.these;
-              ghc-prof = doJailbreak hprev.ghc-prof;
```

Comment:
Some of these overrides are definitely required, and I am currently stuck trying to replicate them with `some-cabal-hashes.nix`